pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -187,20 +186,13 @@ func (a *index) constructService(svc *v1.Service, w *Waypoint) *workloadapi.Serv
 	}
 
 	var lb *workloadapi.LoadBalancing
-	if svc.Spec.TrafficDistribution != nil && *svc.Spec.TrafficDistribution == v1.ServiceTrafficDistributionPreferClose {
-		lb = &workloadapi.LoadBalancing{
-			// Prefer endpoints in close zones, but allow spilling over to further endpoints where required.
-			RoutingPreference: []workloadapi.LoadBalancing_Scope{
-				workloadapi.LoadBalancing_NETWORK,
-				workloadapi.LoadBalancing_REGION,
-				workloadapi.LoadBalancing_ZONE,
-				workloadapi.LoadBalancing_SUBZONE,
-			},
-			Mode: workloadapi.LoadBalancing_FAILOVER,
-		}
+
+	// The TrafficDistribution field is quite new, so we allow a legacy annotation option as well
+	preferClose := svc.Annotations["experimental.istio.io/traffic-distribution"] == "PreferClose"
```

Comment:
Should "PreferClose" be a constant? I think we could also use `v1.ServiceTrafficDistributionPreferClose`

Reply:
Also I guess worth considering, do we want this to be case sensitive?